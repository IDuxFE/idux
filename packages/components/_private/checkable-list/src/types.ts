/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
