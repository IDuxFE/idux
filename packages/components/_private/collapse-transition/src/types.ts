/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
