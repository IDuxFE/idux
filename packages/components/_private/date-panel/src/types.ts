/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export const datePanelProps = {
  cellTooltip: Function as PropType<(cell: { value: Date; disabled: boolean }) => string | void>,
  disabledDate: Function as PropType<(date: Date) => boolean>,
  type: {
    type: String as PropType<DatePanelType>,
    default: 'date',
  },
  value: [Date, Array] as PropType<Date | (Date | undefined)[]>,
  activeDate: Date,
  visible: {
    type: Boolean,
    default: undefined,
  },
  isSelecting: {
    type: Boolean,
    default: undefined,
  },

  onCellClick: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  onCellMouseenter: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  'onUpdate:activeDate': [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
}

export type DatePanelProps = ExtractInnerPropTypes<typeof datePanelProps>
export type DatePanelPublicProps = ExtractPublicPropTypes<typeof datePanelProps>
export type DatePanelComponent = DefineComponent<
  Omit<HTMLAttributes, keyof DatePanelPublicProps> & DatePanelPublicProps
>
export type DatePanelInstance = InstanceType<DefineComponent<DatePanelProps>>

export type DatePanelType = 'date' | 'week' | 'month' | 'quarter' | 'year'

// private
export const panelRowProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
}

export const panelCellProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
  cellIndex: {
    type: Number,
    required: true,
  },
}
