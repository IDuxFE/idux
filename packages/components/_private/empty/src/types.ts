/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { EmptyProps as IxEmptyProps } from '@idux/components/empty'

export interface EmptyProps {
  empty?: string | IxEmptyProps
}
