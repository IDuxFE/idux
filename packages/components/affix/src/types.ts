import type { DefineComponent } from 'vue'

export type Direction = 'top' | 'bottom' | 'left' | 'right'

export type OffsetOpt = {
  [key in Direction]?: number | string
}

export type AffixStyle = {
  position?: 'fixed' | 'absolute'
  top?: string
  bottom?: string
  left?: string
  right?: string
}

/**
 * props类型
 *
 * @interface AffixOriginalProps
 */
interface AffixOriginalProps {
  /**
   * 偏移量
   * type为number时，默认是top
   * top和bottom、left和right同时存在时，top和left优先级较高
   */
  offset: number | string | OffsetOpt
  // 用于定位的容器，会监听容器的滚动事件
  target?: string | HTMLElement | Window
}

export type AffixProps = Readonly<AffixOriginalProps>

export type AffixComponent = InstanceType<DefineComponent<AffixProps>>
