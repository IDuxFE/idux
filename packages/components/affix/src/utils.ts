/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { AffixDirection, AffixOffset } from './types'

import { isObject, isUndefined } from 'lodash-es'

import { convertNumber, isHTMLElement, off, on } from '@idux/cdk/utils'

export type ContentStyle = {
  position?: 'fixed' | 'absolute'
  top?: string
  bottom?: string
  left?: string
  right?: string
  width?: string
  height?: string
}

type AffixDirectionNumber = Record<AffixDirection, number>

const events = ['scroll', 'resize'] as const
const directions = ['top', 'bottom', 'left', 'right'] as const

export function normalizeOffset(offset: AffixOffset): Partial<AffixDirectionNumber> {
  if (!isObject(offset)) {
    return { top: convertNumber(offset) }
  }
  const _offset: Partial<AffixDirectionNumber> = {}
  ;(Object.keys(offset) as AffixDirection[]).forEach(dire => {
    _offset[dire] = convertNumber(offset[dire])
  })
  return _offset
}

/**
 * get target position based on container
 */
export function getTargetRect(target: HTMLElement, container: HTMLElement | Window): AffixDirectionNumber {
  const targetRect = target.getBoundingClientRect()
  const containerRect = isHTMLElement(container)
    ? container.getBoundingClientRect()
    : { top: 0, bottom: window.innerHeight, left: 0, right: window.innerWidth }
  return {
    top: targetRect.top - containerRect.top,
    left: targetRect.left - containerRect.left,
    bottom: containerRect.bottom - targetRect.bottom,
    right: containerRect.right - targetRect.right,
  }
}

export function getTargetSize(target: HTMLElement | Window): Record<'width' | 'height', number> {
  if (target === window) {
    return {
      width: target.innerWidth,
      height: target.innerHeight,
    }
  }
  return {
    width: (target as HTMLElement).offsetWidth,
    height: (target as HTMLElement).offsetHeight,
  }
}

export function observeTarget(target: HTMLElement | Window, cb: (...args: unknown[]) => unknown): void {
  events.forEach(event => {
    on(target, event, cb)
  })
}

export function removeObserveTarget(target: HTMLElement | Window, cb: (...args: unknown[]) => unknown): void {
  events.forEach(event => {
    off(target, event, cb)
  })
}

export function isDireSticky(
  dire: AffixDirection,
  affixRect: AffixDirectionNumber,
  offsetOption: Partial<AffixDirectionNumber>,
): boolean {
  return !isUndefined(offsetOption[dire]) && affixRect[dire] <= offsetOption[dire]!
}

export function isSticky(affixRect: AffixDirectionNumber, offsetOption: Partial<AffixDirectionNumber>): boolean {
  return directions.some(dire => isDireSticky(dire, affixRect, offsetOption))
}

export function calcPosition(
  affixRect: AffixDirectionNumber,
  offsetOption: Partial<AffixDirectionNumber>,
  target: Window | HTMLElement,
): ContentStyle {
  const style: ContentStyle = {}

  if (isSticky(affixRect, offsetOption)) {
    style.position = target === window ? 'fixed' : 'absolute'

    const _directions: AffixDirection[] = [
      isDireSticky('bottom', affixRect, offsetOption) ? 'bottom' : 'top',
      isDireSticky('right', affixRect, offsetOption) ? 'right' : 'left',
    ]

    _directions.forEach(dire => {
      if (isDireSticky(dire, affixRect, offsetOption)) {
        style[dire] = `${offsetOption[dire]! - (target === window ? 0 : affixRect[dire])}px`
      } else {
        style[dire] = `${target === window ? affixRect[dire] : 0}px`
      }
    })
  }
  return style
}
