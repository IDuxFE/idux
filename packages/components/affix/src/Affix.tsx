/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ContentStyle } from './utils'
import type { CSSProperties } from 'vue'

import { computed, defineComponent, nextTick, onMounted, onUnmounted, ref, watch } from 'vue'

import { throttleRAF } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { covertTarget } from '@idux/components/utils'

import { affixProps } from './types'
import {
  calcPosition,
  getTargetRect,
  getTargetSize,
  isSticky,
  normalizeOffset,
  observeTarget,
  removeObserveTarget,
} from './utils'

export default defineComponent({
  name: 'IxAffix',
  props: affixProps,
  setup(props, { slots }) {
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-affix`)
    const contentStyle = ref<ContentStyle>({} as ContentStyle)
    const affixStyle = ref({} as CSSProperties)

    const targetRef = ref<Window | HTMLElement | null>(null)
    const affixRef = ref<HTMLElement | null>(null)
    const contentRef = ref<HTMLElement | null>(null)

    const offset = computed(() => normalizeOffset(props.offset))

    const isStickyRef = ref(false)
    watch(isStickyRef, value => props.onChange?.(value))

    const throttleMeasure = throttleRAF(measure)

    function measure(event?: unknown) {
      if (!affixRef.value || !targetRef.value) {
        return
      }
      const affixRect = getTargetRect(affixRef.value, targetRef.value)
      isStickyRef.value = isSticky(affixRect, offset.value)
      contentStyle.value = calcPosition(affixRect, offset.value, targetRef.value)
      if (isStickyRef.value && contentRef.value) {
        if (event && (event as Event).type === 'resize') {
          clearStyle()

          nextTick(() => {
            measure()
          })

          return
        }

        const { width, height } = getTargetSize(contentRef.value)
        contentStyle.value = {
          ...contentStyle.value,
          width: `${width}px`,
          height: `${height}px`,
        }
        affixStyle.value = {
          width: `${width}px`,
          height: `${height}px`,
        }
        if (targetRef.value !== window) {
          affixStyle.value.position = 'relative'
        }
      }
    }

    function clearStyle() {
      affixStyle.value = {}
      contentStyle.value = {}
    }

    onMounted(() => {
      nextTick(() => {
        initContainer()
        measure()
      })
    })

    onUnmounted(() => {
      removeObserveTarget(targetRef.value!, throttleMeasure)
      throttleMeasure.cancel?.()
    })

    watch(() => props.offset, measure)
    watch(
      () => props.target,
      () => {
        removeObserveTarget(targetRef.value!, throttleMeasure)
        initContainer()
        measure()
      },
    )

    function initContainer() {
      targetRef.value = covertTarget(props.target)
      observeTarget(targetRef.value, throttleMeasure)
    }

    return () => {
      const prefixCls = mergedPrefixCls.value
      return (
        <div ref={affixRef} style={affixStyle.value} class={prefixCls}>
          <div ref={contentRef} class={`${prefixCls}-content`} style={contentStyle.value}>
            {slots.default?.()}
          </div>
        </div>
      )
    }
  },
})
