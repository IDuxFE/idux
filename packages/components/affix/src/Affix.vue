<template>
  <div ref="affixRef" :style="wrapperStyle" class="ix-affix">
    <div ref="contentRef" class="ix-affix-content" :style="affixStyle">
      <slot></slot>
    </div>
  </div>
</template>
<script lang="ts">
import type { CSSProperties } from 'vue'
import type { AffixStyle } from './utils'

import { defineComponent, computed, watch, ref, onMounted, nextTick, onUnmounted } from 'vue'
import { throttleRAF } from '@idux/cdk/utils'
import { getTarget } from '@idux/components/utils'
import { affixProps } from './types'
import {
  getTargetRect,
  getTargetSize,
  calcPosition,
  normalizeOffset,
  isSticky,
  observeTarget,
  removeObserveTarget,
} from './utils'

export default defineComponent({
  name: 'IxAffix',
  props: affixProps,
  setup(props) {
    const affixStyle = ref<AffixStyle>({} as AffixStyle)
    const wrapperStyle = ref({} as CSSProperties)

    const targetRef = ref<Window | HTMLElement | null>(null)
    const affixRef = ref<HTMLElement | null>(null)
    const contentRef = ref<HTMLElement | null>(null)

    const offset = computed(() => normalizeOffset(props.offset))

    const isStickyRef = ref(false)
    watch(isStickyRef, value => props.onChange?.(value))

    const throttleMeasure = throttleRAF(measure)

    function measure() {
      if (!affixRef.value || !targetRef.value) {
        return
      }
      const affixRect = getTargetRect(affixRef.value, targetRef.value)
      isStickyRef.value = isSticky(affixRect, offset.value)
      affixStyle.value = calcPosition(affixRect, offset.value, targetRef.value)
    }

    onMounted(() => {
      nextTick(() => {
        initWrapper()
        initContainer()
        measure()
      })
    })

    onUnmounted(() => {
      removeObserveTarget(targetRef.value!, throttleMeasure)
      throttleMeasure.cancel?.()
    })

    watch(() => props.offset, measure)
    watch(
      () => props.target,
      () => {
        removeObserveTarget(targetRef.value!, throttleMeasure)
        initContainer()
        measure()
      },
    )

    function initWrapper() {
      if (!contentRef.value) {
        return
      }
      const { width, height } = getTargetSize(contentRef.value)
      wrapperStyle.value = {
        position: 'relative',
        width: `${width}px`,
        height: `${height}px`,
      }
    }

    function initContainer() {
      targetRef.value = getTarget(props.target)
      observeTarget(targetRef.value, throttleMeasure)
    }

    return {
      affixRef,
      contentRef,
      affixStyle,
      wrapperStyle,
    }
  },
})
</script>
