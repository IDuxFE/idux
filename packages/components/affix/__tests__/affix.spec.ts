import { mount, MountingOptions } from '@vue/test-utils'
import { nextTick, ref } from 'vue'
import { wait, renderWork } from '@tests'
import IxAffix from '../src/Affix.vue'
import { AffixOffset, AffixProps } from '../src/types'

interface Offset {
  top: number
  right: number
  bottom: number
  left: number
}

let rectMap = {} as { container: Offset; 'ix-affix': Offset }

const affixSize = {
  width: 100,
  height: 100,
}

const containerSize = {
  width: 300,
  height: 300,
}

function initRectMap() {
  rectMap = {
    'ix-affix': {
      top: 100,
      right: 100 + affixSize.width,
      bottom: 100 + affixSize.height,
      left: 100,
    },
    container: {
      top: 50,
      right: 50 + containerSize.width,
      bottom: 50 + containerSize.height,
      left: 50,
    },
  }
}

async function scrollTarget(x: number, y: number, target: Window | Element = window, timeout = 200): Promise<unknown> {
  initRectMap()
  const cls = ['ix-affix']
  target === window && cls.push('container')
  ;(cls as ['ix-affix', 'container']).forEach(item => {
    rectMap[item].top -= y
    rectMap[item].bottom -= y
    rectMap[item].left -= x
    rectMap[item].right -= x
  })
  target.dispatchEvent(new Event('scroll'))
  return wait(timeout)
}

describe('Affix.vue', () => {
  const AffixMount = (options?: MountingOptions<Partial<AffixProps>>) => mount(IxAffix, { ...options })

  beforeEach(() => {
    initRectMap()

    const spyGetBoundingClientRect = function (this: HTMLElement) {
      const className = this.className as 'ix-affix' | 'container'
      return (rectMap[className] || {}) as DOMRect
    }

    jest.spyOn(HTMLElement.prototype, 'getBoundingClientRect').mockImplementation(spyGetBoundingClientRect)
  })

  renderWork(IxAffix)

  test('offset work', async () => {
    const wrapper = AffixMount({ props: { offset: 0 }, slots: { default: 'Test Affix' } })

    await nextTick()
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(10, 10)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(10, 100)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(100, 100)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(110, 110)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ offset: { left: 0 } })

    await scrollTarget(10, 10)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(100, 10)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(100, 100)
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(110, 110)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ offset: { top: 0, left: 0 } })
    await scrollTarget(110, 110)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ offset: { bottom: 0, right: 0 } })
    await scrollTarget(-window.innerWidth + affixSize.width + 100, -window.innerHeight + affixSize.height + 100)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('target work', async () => {
    const offset = ref<AffixOffset>(50)
    const target = ref<HTMLElement | undefined>(undefined)
    const wrapper = mount(
      {
        components: { IxAffix },
        template: `<div class="container"><ix-affix :offset="offset" :target="target">Test Affix</ix-affix></div>`,
        setup() {
          return { offset, target }
        },
      },
      {
        attachTo: document.body,
      },
    )

    expect(wrapper.html()).toMatchSnapshot()

    target.value = document.querySelector('.container') as HTMLElement
    await nextTick()
    expect(wrapper.html()).toMatchSnapshot()

    await scrollTarget(50, 50, target.value)
    expect(wrapper.html()).toMatchSnapshot()

    offset.value = { top: 50, left: 50 }

    await scrollTarget(100, 100, target.value)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emit work', async () => {
    const onChange = jest.fn()

    const wrapper = AffixMount({ props: { offset: 0 }, attrs: { onChange }, slots: { default: 'Test Affix' } })

    await nextTick()
    expect(onChange).not.toBeCalled()

    await scrollTarget(100, 100)
    expect(onChange).toBeCalledWith(true)

    await scrollTarget(0, 0)
    expect(onChange).toBeCalledWith(false)

    wrapper.setProps({ offset: 100 })
    await nextTick()
    expect(onChange).toBeCalledWith(true)
  })
})
