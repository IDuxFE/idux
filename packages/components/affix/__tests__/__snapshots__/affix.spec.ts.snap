// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Affix.vue offset work 1`] = `"<div class=\\"ix-affix\\"><div class=\\"ix-affix-content\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 2`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 3`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 0px; left: 90px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 4`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 0px; left: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 5`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 0px; left: -10px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 6`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 7`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 90px; left: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 8`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 0px; left: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 9`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: -10px; left: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 10`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; top: 0px; left: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue offset work 11`] = `"<div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: fixed; bottom: 0px; right: 0px;\\">Test Affix</div></div>"`;

exports[`Affix.vue render work 1`] = `"<div class=\\"ix-affix\\"><div class=\\"ix-affix-content\\"></div></div>"`;

exports[`Affix.vue target work 1`] = `"<div class=\\"container\\"><div class=\\"ix-affix\\"><div class=\\"ix-affix-content\\">Test Affix</div></div></div>"`;

exports[`Affix.vue target work 2`] = `"<div class=\\"container\\"><div class=\\"ix-affix\\"><div class=\\"ix-affix-content\\" style=\\"position: absolute; top: 0px; left: 0px;\\">Test Affix</div></div></div>"`;

exports[`Affix.vue target work 3`] = `"<div class=\\"container\\"><div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: absolute; top: 50px; left: 0px;\\">Test Affix</div></div></div>"`;

exports[`Affix.vue target work 4`] = `"<div class=\\"container\\"><div class=\\"ix-affix\\" style=\\"position: relative; width: 0px; height: 0px;\\"><div class=\\"ix-affix-content\\" style=\\"position: absolute; top: 100px; left: 100px;\\">Test Affix</div></div></div>"`;
